Dockerfile

```diff
@@ -1,20 +1,53 @@
-FROM node:10-alpine
+FROM alpine:3.10 AS buildcontainer
 
-RUN apk update && apk upgrade && apk add nmap nmap-scripts
+ENV  NMAP_SHA256SUM="fcfa5a0e42099e12e4bf7a68ebe6fde05553383a682e816a7ec9256ab4773faa" \
+     NMAP_VERSION=7.80
 
-COPY package.json package-lock.json /src/
+WORKDIR /nmap
 
-WORKDIR /src
+ARG NMAP_PACKAGE="nmap-${NMAP_VERSION}.tar.bz2"
+ARG NMAP_URI="https://nmap.org/dist/${NMAP_PACKAGE}"
+
+RUN echo "Installing Nmap ${NMAP_VERSION}" && \
+    apk update && \
+    apk upgrade && \
+    apk add build-base && \
+    apk add flex && \
+    apk add libcap-dev && \
+    apk add openssl-dev && \
+    apk add libssh2-dev && \
+    apk add bison && \
+    apk add curl && \
+    set -ex && \
+    curl -fsSLO ${NMAP_URI} && \
+    echo "${NMAP_SHA256SUM}  ${NMAP_PACKAGE}" | sha256sum -c - && \
+    bzip2 -cd "${NMAP_PACKAGE}" | tar xvf - && \
+    cd "nmap-${NMAP_VERSION}" && \
+    ./configure && \
+    make -s -j "$(nproc)" && \
+    make -s install
 
-RUN ls
+FROM node:12-alpine
 
-RUN npm install --production
+ARG NMAP_VERSION=7.80
 
+COPY package.json package-lock.json /src/ 
+COPY --from=buildcontainer /usr/local/ /usr/local
 COPY . /src
 
-HEALTHCHECK --interval=30s --timeout=5s --start-period=120s --retries=3 CMD node healthcheck.js || exit 1
+WORKDIR /src
 
-RUN addgroup -S nmap_group && adduser -S -g nmap_group nmap_user 
+RUN apk update && \
+    apk upgrade && \
+    apk add libssh2 && \
```

Comment:
In the actual image sections it is a best practice to install apk dependencies with `the --no-cache` flag. Otherwise the alpine package manager might store a cache in the image which increases the size but doesn't help with anything.